crates/iota-indexer/src/types.rs

```diff
@@ -157,10 +205,10 @@ impl IndexedEpochInfo {
             ),
             last_checkpoint_id: Some(*last_checkpoint_summary.sequence_number()),
             epoch_end_timestamp: Some(last_checkpoint_summary.timestamp_ms),
-            storage_charge: Some(event.storage_charge),
```

Comment:
Couldn't we just access these fields, and skip the match?

```rust
storage_charge: Some(event.storage_charge()),
```

Reply:
you mean by adding getters to the `SystemEpochInfoEvent` enum? Do you think this would be more idiomatic/efficient? It certainly seems cleaner.

Reply:
I do think it could be significantly cleaner. A lot of times patterns like this will use `enum_dispatch` and a unifying trait. Alternatively, you could have functions like this:

```rust
impl SystemEpochInfoEvent {
  pub fn storage_charge(&self) -> Option<u64> {
    match self {
      ...
    }
  }
```

Not sure if it's worth it, but it would definitely clean this particular code up

Reply:
I also think it is worth it! @cyberphysic4l check how the [IotaSystemState works](https://github.com/iotaledger/iota/blob/a822eb2d1f1dc1ad7ccc4517e36413e7cc9933e7/crates/iota-types/src/iota_system_state/mod.rs#L175), it's pretty much the same scenario.